This patch allows pfn_valid from linux/asm-sh/page.h to be used in modules,
required by a driver's mmap nopage handler. It is an almost exact copy of 
the IA64 code, except the config option it is conditional on is the same as
that used to conditionally define pfn_valid on the SH.

It also exports fb_destroy_modelist for stmfb.

Signed-off by: Andre' Draszik <andre.draszik@st.com>
Signed off by: Stephen Gallimore <Stephen.Gallimore@st.com>

Index: linux/arch/sh/kernel/sh_ksyms.c
===================================================================
--- linux.orig/arch/sh/kernel/sh_ksyms.c	2007-10-29 12:30:58.144098000 +0000
+++ linux/arch/sh/kernel/sh_ksyms.c	2007-10-29 12:47:57.919270000 +0000
@@ -44,6 +44,14 @@
 EXPORT_SYMBOL(memmove);
 EXPORT_SYMBOL(__copy_user);
 
+#ifdef CONFIG_FLATMEM
+#include <linux/bootmem.h>
+EXPORT_SYMBOL(min_low_pfn);     /* defined by bootmem.c, but not exported by gen
+eric code */
+EXPORT_SYMBOL(max_low_pfn);     /* defined by bootmem.c, but not exported by gen
+eric code */
+#endif
+
 #ifdef CONFIG_MMU
 EXPORT_SYMBOL(get_vm_area);
 #endif
Index: linux/drivers/video/modedb.c
===================================================================
--- linux.orig/drivers/video/modedb.c	2007-10-29 12:48:17.953450000 +0000
+++ linux/drivers/video/modedb.c	2007-10-29 12:49:16.794979000 +0000
@@ -938,6 +938,7 @@
 		kfree(pos);
 	}
 }
+EXPORT_SYMBOL_GPL(fb_destroy_modelist);
 
 /**
  * fb_videomode_to_modelist: convert mode array to mode list
